pkg/v1/tkg/client/management_components.go

```diff
@@ -92,6 +121,27 @@ func (c *TkgClient) getUserConfigVariableValueMap() (map[string]string, error) {
 	return c.GetUserConfigVariableValueMap(path, c.TKGConfigReaderWriter())
 }
 
+func (c *TkgClient) getUserConfigVariableValueMapForUpgrade(kubeconfig, kubecontext string) (map[string]string, error) {
+	clusterClient, err := clusterclient.NewClient(kubeconfig, kubecontext, clusterclient.Options{})
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get cluster client")
+	}
+
+	var tkgPackageConfig managementcomponents.TKGPackageConfig
+
+	bytes, err := clusterClient.GetSecretValue(fmt.Sprintf(tkgpackagedatamodel.SecretName, constants.TKGManagementPackageInstallName, constants.TkgNamespace), constants.TKGPackageValuesFile, constants.TkgNamespace, nil)
```

Comment:
What if the cluster is a pre-pblcm one?

Reply:
Good catch. I think we need to think a bit more about the upgrades from legacy management cluster to new `pblcm` management cluster and how to get the configuration to deploy TKG composite packages from legacy cluster.

Reply:
please file an issue and add a TODO comment here to reference it, as it is something we will definitely need to account for as well.

Reply:
Updated https://github.com/vmware-tanzu/tanzu-framework/issues/2147 to track this work as well.